gnucash/import-export/bi-import/dialog-bi-import.c

```diff
@@ -392,6 +371,28 @@ gnc_bi_import_fix_bis (GtkListStore * store, guint * fixed, guint * deleted,
 
         if (!row_deleted)
             valid = gtk_tree_model_iter_next (GTK_TREE_MODEL (store), &iter);
+        if (valid) gtk_tree_model_get (GTK_TREE_MODEL (store), &iter, ID, &id, -1);
+
+        // If this row was the last row of the invoice...
+        if (!valid || (valid && g_strcmp0 (id, running_id->str) != 0))
+        {
+            on_first_row_of_invoice = TRUE;
+        }
+        else
+        {
+            // The next row has the same id, so it is not the first row of an invoice.
+            // But if the first row of an invoice was deleted, the next row is again the first row of the invoice.
```

Comment:
> If the first row of an invoice was deleted, the next row is again the first row of the invoice

... except you may have lost the header data as not all lines should be required to carry it. You may have to cater for that case by setting the missing data on this new row based on what you had on the removed one.